modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java

```diff
@@ -1403,7 +1403,11 @@ private void replayUpdates(long lowerRevision, long upperRevision) {
                 updatedEntries.add(entry(rocksKeyToBytes(rocksKey), revision, bytesToValue(rocksValue)));
             }
 
-            checkIterator(it);
+            try {
+                checkIterator(it);
```

Comment:
This is about metastorage errors, not storage errors, and should be done in a separate ticket.

Reply:
What should I do with it now? `checkIterator` throws a `RocksDbException` and it needs to be handled

Reply:
You can leave the static method as is, and where necessary, handle the exception correctly.

Reply:
Do you mean catching the `IgniteInternalException`? Why? Every method in this class already throws `MetaStorageException`, what's wrong with the current approach?

Reply:
I think that for metastorage this needs to be fixed separately and not in this ticket, leave it as is for now.

Reply:
jesus christ

Reply:
We can leave `RocksUtils#checkIterator` as it was before the change. We'll do it well in metastorage and delete it then.

Reply:
I don't want to catch `IgniteInternalException` explicitly, it looks like a hack. Since we are going to handle this exception in different ways everywhere, I prefer to propagate the original `RocksException.

Reply:
Okay, let's make them metastorage exceptions then.